src/pynwb/__init__.py

```diff
@@ -535,6 +535,71 @@ def read_nwb(**kwargs):
 
         return nwbfile
 
+@docval({'name': 'path', 'type': (str, Path), 'doc': 'the path to the nwbfile'}, 
+        is_method=False)
+def read_nwb(**kwargs):
+    """Read an NWB file from a local path.
+
+    High-level interface for reading NWB files. Automatically handles both HDF5 
+    and Zarr formats. For advanced use cases (parallel I/O, custom namespaces), 
+    use NWBHDF5IO or NWBZarrIO.
+
+    Parameters
+    ----------
+    path : str or pathlib.Path
+        Path to the NWB file. Can be either a local filesystem path to an HDF5 (.nwb) 
+        or Zarr (.zarr) file
+
+    Returns
+    -------
+    pynwb.NWBFile
+        The loaded NWB file object.
+
+    See Also
+    --------
+    pynwb.NWBHDF5IO : Core I/O class for HDF5 files with advanced options.
+    hdmf_zarr.nwb.NWBZarrIO : Core I/O class for Zarr files with advanced options.
+
+    Notes
+    -----
+    This function uses the following defaults:
+    * Always opens in read-only mode
+    * Automatically loads namespaces
+    * Detects file format based on extension
```

Comment:
```suggestion
    * Detects file format based on whether supported backend IO classes can read the file
```

I think in the current implementation other extensions will work since `can_read` and `IO.read_nwb` methods just check whether the IO can successfully open the file?

Reply:
This lien was strange, I changed to what I intended which is that it reads both zarr and nwb but this is unrelated to the extension of the file.